sympy/stats/crv_types.py

```diff
@@ -1582,6 +1596,9 @@ def _cdf(self, x):
         mu, s = self.mu, self.s
         return S.One/(1 + exp(-(x - mu)/s))
 
+    def _characteristic_function(self, t):
+        return exp(I*t*self.mu) * pi*self.s*t / sinh(pi*self.s*t)
```

Comment:
This should be Piecewise to handle the singularity at 0. Consider
```
X = Logistic("x", 0, 1)
characteristic_function(X)(0)   # nan
```
I recommend having the generic case first: `Piecewise((current_formula, Ne(t, 0)), (S.One, True))`